dynawo/sources/Models/CPP/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.h

```diff
@@ -0,0 +1,410 @@
+  *
+  * Get the discrete variables' value depending on current simulation instant and
+  * current state variables values.
+  *
+  * @param t Simulation instant
+  */
+  void evalZ(const double & t);
+
+ private:
+  /**
+  * @brief filtered voltage amplitude
+  * @return value
+  */
+  double UfRaw() const;
+  /**
+  * @brief bounded filtered voltage amplitude
+  * @return value
+  */
+  double Uf() const;
+  /**
+  * @brief time derivative of filtered voltage amplitude
+  * @return value
+  */
+  double UfRawPrim() const;
+  /**
+  * @brief get real value of voltage
+  * @return value
+  */
+  double Ur() const;
+  /**
+  * @brief get imaginary value of voltage
+  * @return value
+  */
+  double Ui() const;
+  /**
+  * @brief get real value of current
+  * @return value
+  */
+  double Ir() const;
+  /**
+  * @brief get imaginary value of current
+  * @return value
+  */
+  double Ii() const;
+  /**
+  * @brief voltage value
+  * @return value
+  */
+  double U() const;
+  /**
+  * @brief square voltage value
+  * @return value
+  */
+  double U2() const;
+  /**
+  * @brief get the active power in p.u. (SNREF)
+  * @return value
+  */
+  double P() const;
+  /**
+  * @brief get the reactive power in p.u. (SNREF)
+  * @return value
+  */
+  double Q() const;
+  /**
+  * @brief derivative of P with respect to Ur
+  * @param ur real part of the voltage
+  * @param ui imaginary part of the voltage
+  * @param U voltage
+  * @param U2 voltage square
+  * @return value
+  */
+  double P_dUr(const double& ur, const double& ui, const double& U, const double& U2) const;
+  /**
+  * @brief derivative of P with respect to Ui
+  * @param ur real part of the voltage
+  * @param ui imaginary part of the voltage
+  * @param U voltage
+  * @param U2 voltage square
+  * @return value
+  */
+  double P_dUi(const double& ur, const double& ui, const double& U, const double& U2) const;
+  /**
+  * @brief derivative of P with respect to Uf
+  * @param U voltage
+  * @return value
+  */
+  double P_dUf(const double& U) const;
+  /**
+  * @brief derivative of Q with respect to Ur
+  * @param ur real part of the voltage
+  * @param ui imaginary part of the voltage
+  * @param U voltage
+  * @param U2 voltage square
+  * @return value
+  */
+  double Q_dUr(const double& ur, const double& ui, const double& U, const double& U2) const;
+  /**
+  * @brief derivative of Q with respect to Ui
+  * @param ur real part of the voltage
+  * @param ui imaginary part of the voltage
+  * @param U voltage
+  * @param U2 voltage square
+  * @return value
+  */
+  double Q_dUi(const double& ur, const double& ui, const double& U, const double& U2) const;
+  /**
+  * @brief derivative of Q with respect to Ui
+  * @param U voltage
+  * @return value
+  */
+  double Q_dUf(const double& U) const;
+
+  State connectionState_;  ///< "internal" load connection status
+  State connectionState_prev;  ///< "internal" load connection status at previous timestamp
```

Comment:
To improve consistency in notation with Modelica, perhaps call it preConnectionState instead of connectionState_prev